src/multi_vector_simulator/E1_process_results.py

```diff
@@ -677,39 +678,57 @@ def convert_components_to_dataframe(dict_values):
 
     """
 
+    # Read the subdicts energyProduction, energyConversion and energyStorage as separate dicts
     components1 = dict_values[ENERGY_PRODUCTION]
     components2 = dict_values[ENERGY_CONVERSION]
+    components3 = dict_values[ENERGY_STORAGE]
 
+    # Read the keys of the above dicts into separate lists
     comp1_keys = list(components1.keys())
     comp2_keys = list(components2.keys())
+    comp3_keys = list(components3.keys())
 
+    # Add the above dictionaries and lists of keys into new lists for iterating through, later
+    comp_dict_list = [components1, components2]
+    components_list = [comp1_keys, comp2_keys]
+
+    # Dict to hold the data for creating a pandas dataframe
     components = {}
-    # Defining the columns of the table to be printed
-    for comps in comp1_keys:
-        components.update(
-            {
-                comps: [
-                    components1[comps][OEMOF_ASSET_TYPE],
-                    components1[comps][ENERGY_VECTOR],
-                    components1[comps][UNIT],
-                    components1[comps][INSTALLED_CAP][VALUE],
-                    components1[comps][OPTIMIZE_CAP][VALUE],
-                ]
-            }
-        )
-    for comps in comp2_keys:
+
+    # Defining the columns of the table and filling them up with the appropriate data
+    for (component_key, comp_dict) in zip(components_list, comp_dict_list):
+        for comps in component_key:
+            components.update(
+                {
+                    comps: [
+                        comp_dict[comps][OEMOF_ASSET_TYPE],
+                        comp_dict[comps][ENERGY_VECTOR],
+                        comp_dict[comps][UNIT],
+                        comp_dict[comps][INSTALLED_CAP][VALUE],
+                        comp_dict[comps][OPTIMIZE_CAP][VALUE],
+                    ]
+                }
+            )
+
+    for storage_component in comp3_keys:
+        comp_label = components3[storage_component]["output power"]["label"]
         components.update(
             {
-                comps: [
-                    components2[comps][OEMOF_ASSET_TYPE],
-                    components2[comps][ENERGY_VECTOR],
-                    components2[comps][UNIT],
-                    components2[comps][INSTALLED_CAP][VALUE],
-                    components2[comps][OPTIMIZE_CAP][VALUE],
+                comp_label: [
+                    components3[storage_component][OEMOF_ASSET_TYPE],
+                    components3[storage_component][ENERGY_VECTOR],
+                    components3[storage_component]["output power"]["installedCap"][
+                        "unit"
+                    ],
+                    components3[storage_component]["output power"][INSTALLED_CAP][
+                        VALUE
+                    ],
+                    components3[storage_component]["output power"][OPTIMIZE_CAP][VALUE],
```

Comment:
You should do a loop here though `[INPUT_POWER, STORAGE_CAPACITY, OUTPUT_POWER]` as all three should be in the report. Please use constant variables ;)